editors/code/src/commands/on_enter.ts

```diff
@@ -10,22 +10,15 @@ async function handleKeypress(ctx: Ctx) {
 
     if (!editor || !client) return false;
 
-    const request: lc.TextDocumentPositionParams = {
+    const change = await client.sendRequest(ra.onEnter, {
         textDocument: { uri: editor.document.uri.toString() },
         position: client.code2ProtocolConverter.asPosition(
             editor.selection.active,
         ),
-    };
-    const change = await client.sendRequest<undefined | SourceChange>(
-        'rust-analyzer/onEnter',
-        request,
-    ).catch(
-        (_error: any) => {
-            // FIXME: switch to the more modern (?) typed request infrastructure
-            // client.logFailedRequest(OnEnterRequest.type, error);
-            return Promise.resolve(null);
```

Comment:
FYI: @matklad  there is no need to wrap the return value into a promise in the `then(onfulfilled, onrejected)/catch(onrejected)` handlers.